packages/@ember/-internals/glimmer/tests/integration/helpers/invoke-helper-test.js

```diff
@@ -0,0 +1,272 @@
+        let count = 0;
+        let helper;
+
+        class PlusOneHelper extends Helper {
+          @tracked number = 4;
+
+          constructor() {
+            super(...arguments);
+            helper = this;
+          }
+
+          compute() {
+            count++;
+            return this.number + 1;
+          }
+        }
+
+        class PlusOne {
+          plusOne = invokeHelper(this, PlusOneHelper);
+
+          get value() {
+            return getValue(this.plusOne);
+          }
+        }
+
+        let instance = new PlusOne();
+
+        assert.equal(instance.value, 5, 'helper works');
+        assert.equal(instance.value, 5, 'helper works');
+        assert.equal(count, 1, 'helper only called once');
+
+        helper.number = 5;
+
+        assert.equal(instance.value, 6, 'helper works');
+        assert.equal(count, 2, 'helper called a second time');
+      }
+
+      ['@test helper destroys correctly when context object is destroyed'](assert) {
+        assert.expect(3);
+
+        let context = {};
+
+        class TestHelper extends Helper {
+          willDestroy() {
+            assert.ok(true, 'helper destroyed');
+          }
+        }
+
+        let helper = invokeHelper(context, TestHelper);
+
+        runTask(() => destroy(context));
+
+        assert.ok(isDestroyed(context), 'context destroyed');
+        assert.ok(isDestroyed(helper), 'helper cache destroyed');
+      }
+
+      ['@test helper destroys correctly when helper cache is destroyed'](assert) {
+        assert.expect(3);
+
+        let context = {};
+
+        class TestHelper extends Helper {
+          willDestroy() {
+            assert.ok(true, 'helper destroyed');
+          }
+        }
+
+        let helper = invokeHelper(context, TestHelper);
+
+        runTask(() => destroy(helper));
+
+        assert.notOk(isDestroyed(context), 'context NOT destroyed');
+        assert.ok(isDestroyed(helper), 'helper cache destroyed');
+      }
+
+      ['@test throws an error if value is accessed after it is destroyed']() {
+        expectAssertion(() => {
+          let helper = invokeHelper({}, class extends Helper {});
+
+          runTask(() => destroy(helper));
+
+          getValue(helper);
+        }, /You attempted to get the value of a helper after the helper was destroyed, which is not allowed/);
+      }
+
+      ['@test asserts if no context object is passed']() {
+        expectAssertion(() => {
+          invokeHelper(undefined, class extends Helper {});
+        }, /Expected a context object to be passed as the first parameter to invokeHelper, got undefined/);
+      }
+
+      ['@test asserts if no manager exists for the helper definition']() {
+        expectAssertion(() => {
+          invokeHelper({}, class {});
+        }, /Expected a helper definition to be passed as the second parameter to invokeHelper, but no helper manager was found. Did you use setHelperManager to associate a helper manager?/);
```

Comment:
As mentioned above, this message should be augmented to include _some_ information RE: the helper that was passed.